.travis.yml

```
@@ -45,6 +48,7 @@ before_script:
   - PWNLIB_NOTERM=1 python -c 'from pwn import *; print(pwnlib.term.term_mode)'
   - PWNLIB_NOTERM=1 python -c 'from pwn import *; print(pwnlib.term.term_mode)'
   - PWNLIB_NOTERM=1 python -c 'from pwn import *; print(pwnlib.term.term_mode)'
+  - sudo sh -c "echo 0 > /proc/sys/kernel/yama/ptrace_scope"
```

This shouldn't be necessary, as we do some prctl() in pwntools to mitigate this already

---

That's necessary. `prctl()` only affects direct child process.
You can try this by yourself:
https://github.com/Gallopsled/pwntools/blob/0491972d1ca79dbdd85f8a021509210bdb53f10c/pwnlib/gdb.py#L587-L597
The `attach` should fail if `ptrace_scope` is not set to `0`.

---

I see, the process is created by `socat`, not us. This is bad, because pwntools users won't be able to do this either without disabling Yama.

---

I think most users won't use pwntools in this way, so it's okey to leave this problem here😃